### ISSUE TITLE
Moonshine ant build script is failing on Windows

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#379]
Ant script is failing during the build on target [CompileSWFWindowsSignedPackage](https://github.com/prominic/Moonshine-IDE/blob/0a25a328edaac522efcde412243f97630681a0b6/ide/MoonshineSharedCore/build/build-lib.xml#L255). Script seems to do not have problem with building swf, but rather signing later. 

```
CompileSWFWindowsSelfSignedPackage:
     [echo] Self-signed Process Running
     [java] unexpected failure: Platform conversion failed. Process exited with error 5. Output was "Conversion complete
     [java] [ErrorEvent type="error" bubbles=false cancelable=false eventPhase=2 text="Error #2038" errorID=2038]
     [java] "
     [java] java.io.IOException: Platform conversion failed. Process exited with error 5. Output was "Conversion complete
     [java] [ErrorEvent type="error" bubbles=false cancelable=false eventPhase=2 text="Error #2038" errorID=2038]
     [java] "
     [java]     at com.adobe.air.nai.NativePackager.conversionFail(NativePackager.java:279)
     [java]     at com.adobe.air.nai.NativePackager.invokePlatformConversion(NativePackager.java:260)
     [java]     at com.adobe.air.nai.NativePackager.createPackage(NativePackager.java:133)
     [java]     at com.adobe.air.ADT.parseArgsAndGo(ADT.java:696)
     [java]     at com.adobe.air.ADT.run(ADT.java:485)
     [java]     at com.adobe.air.ADT.main(ADT.java:535)
     [java] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
```

Cause of that problem is following [dll](https://github.com/prominic/Moonshine-IDE/tree/master/ide/MoonshineSharedCore/src/elements/jdt-language-server/config_win/org.eclipse.equinox.launcher/org.eclipse.equinox.launcher.win32.win32.x86_1.1.800.v20180611-1436) which required by language server.

I have prepared build which produce installation package from Intellij IDE and it's working without fail.

**Comment by rat-moonshine** [comment]
I'm not sure, but as a probability this may possible in IntelliJ - they prepares only  the bundle captive runtime first, then use their any own packager to prepare the bundle as .EXE. In our case, we uses the SDK's own 'adt' to prepare the .EXE (which fails for some reason).

**Comment by piotrzarzycki21** [comment]
Can you try to prepare installer using Flash Builder ? Is it possible ?

**Comment by rat-moonshine** [comment]
I spent some more time investigating this, and this is more like a name-length/naming issue than _.dll_ issue to me.

I tried to rename the _.dll_ file prior to bundle, but packaging fails with same error. I thought its the bytecode writing which fails against the renamed file some way.

I deleted the _.dll_ file and tried an Ant build which finish successful. But I noticed that 'org.eclipse.equinox.launcher' (the folder where _.dll_ file was) is completely missing when installed from the package (maybe it didn't bundled because the folder was empty).

Then I shorten the name of the folder that contains the _.dll_ file from "org.eclipse.equinox.launcher.win32.win32.x86_1.1.800.v20180611-1436" (67 chars) to "org.eclipse.equinox.launcher.win32" (34 chars); That completes the packaging successfully. I also see requisite files (including _.dll_) placed properly when installed from the build. 

I couldn't found any strong-point on this file naming limitation while Ant building online, though. This may also possible that any/certain positioning of characters in the folder name "org.eclipse.equinox.launcher.win32.win32.x86_1.1.800.v20180611-1436" causing some problem - but that requires to play more with this.

Flash Builder bundled the packaging without any problem, though; But I don't know what its internal processese are.

**Comment by piotrzarzycki21** [comment]
I may figure it out what is happening. I will provide details once confirm my solution later today.

**Comment by joshtynjala** [comment]
Let me know if commit 7b41f7c69028e558886b55b6a4a5236580310f48 fixes this issue.

While attempting to rename the problematic *.dll* file, I discovered that the file with the original name kept appearing again, even after I deleted it. I re-downloaded the original package *jdt-language-server-latest.tar.gz* for the Java language server, and I saw that *config_win* contains only *config.ini* in a fresh download. This led me to realize that the other files in *config_win* are supposed to be automatically populated when the Java language server starts up. I must have committed the contents of *config_win* after I had already run the Java language server at least one time.

I have removed everything in *config_win* except *config.ini*, and I'm seeing that the other files are automatically created when the Java language server starts up. Hopefully, this works correctly when Moonshine is installed and not just running in the debugger. Otherwise, we may have to take an additional step of copying the *jdt-language-server* files somewhere that the app has write access. We'd also need to find a way to keep these files up to date when we want to upgrade the language server to a new version.
